src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -1291,16 +1291,15 @@ private void updateStudyDbId(BrAPIStudy study, String programKey) {
     private void updateGermplasmDbId(BrAPIGermplasm germplasm) {
         this.observationUnitByNameNoScope.values()
                                          .stream()
-                                         .filter(obsUnit -> obsUnit.getBrAPIObject()
-                                                                   .getGermplasmName() != null &&
-                                                 obsUnit.getBrAPIObject()
-                                                        .getGermplasmName()
-                                                        .equals(germplasm.getGermplasmName()))
+                                         .filter(obsUnit -> germplasm.getAccessionNumber() != null &&
+                                                 germplasm.getAccessionNumber().equals(obsUnit
+                                                         .getBrAPIObject()
+                                                         .getAdditionalInfo().getAsJsonObject()
```

Comment:
Do we need a migration to add gids to existing observation units additionalinfo?

Reply:
Most existing OUs have the correct gemplasm dbId assigned, so a migration adding the GID wouldn't matter either way for them. It's a question more for Alex and Shawn to estimate how many cases of existing OUs using germplasm there are. They haven't com across this until now, so I'm guessing it's a small number where manually fixing it in the species database would make more sense. I'm not sure what a migration would look like. It wouldn't be a case of just adding the GID to the OU additional info since the existing id in the OU germplasmDbId field could be wrong. You couldn't easily tell if it was right or wrong, and if it's wrong, it's not clear which germplasmDbId would be correct? So a migration just looks like re-importing the experiment with bug fix in place.